File: ccif/src/java/other/org/himalay/ccif/ResetPin.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:05 EST 2016")

public  class ResetPin extends BinStruct implements PublicBinMsg{




    // members variables
    // resetpinControl
    public short resetpinControl ;

        public ResetPin () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize resetpinControl
        
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read resetpinControl
        {resetpinControl=(short)(istream.readUnsignedByte()); retVal+=1;}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
    
                // write resetpinControl
        ostream.writeByte(resetpinControl); retVal +=1;
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("ResetPin\n");
    dc.increaseIndent();
        int retVal= 0;
        // write resetpinControl
        dc.indent();dc.getPs().println("resetpinControl="+resetpinControl+"(0x"+ Integer.toHexString(resetpinControl) + ")" );
dc.decreaseIndent();
        return retVal;
    }


        // Getter for resetpinControl
    //public short getResetpinControl()
    //{
    //    return resetpinControl ;
    //}

    
    // Setter for resetpinControl
    //public void setResetpinControl(short val)
    //{
    //    this.resetpinControl= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/ResetPin.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ResetPin.h"

#ifdef org_himalay_ccif_ResetPin__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ResetPin::ResetPin() {
    references= 0;


    // resetpinControl
    resetpinControl= (ui8)0;
}
// Destructor
ResetPin::~ResetPin() {
    // resetpinControl

}

int ResetPin::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // resetpinControl
    {
        resetpinControl=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int ResetPin::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read resetpinControl
    {
        resetpinControl=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int ResetPin::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write resetpinControl
    ostream.writeUI8(resetpinControl);
    retVal +=1;

    return retVal;
}

int ResetPin::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ResetPin\n");
    dc.increaseIndent();
    int retVal= 0;
    // write resetpinControl
    dc.indent();
    dc.getPs().print("resetpinControl=");
    dc.getPs().println((long)resetpinControl);
    dc.decreaseIndent();
    return retVal;
}


// Getter for resetpinControl
//ui8 ResetPin::getResetpinControl()
//{
//return this->resetpinControl;
//}

// Setter for resetpinControl
//void ResetPin::setResetpinControl(ui8 val)
//{
//this->resetpinControl= val;
//}

int ResetPin::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_ResetPin__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::ResetPin* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::ResetPin* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code